azurelinuxagent/ga/exthandlers.py

```diff
@@ -362,9 +360,37 @@ def handle_ext_handlers(self, etag=None):
 
         self.ext_handlers.extHandlers.sort(key=operator.methodcaller('sort_key'))
         for ext_handler in self.ext_handlers.extHandlers:
-            # TODO: handle install in sequence, enable in parallel
+            self.wait_on_ext_handler_dependencies(ext_handler)
             self.handle_ext_handler(ext_handler, etag)
-    
+
+    def wait_on_ext_handler_dependencies(self, ext_handler):
+        dependencies = sum([e.dependencies for e in ext_handler.properties.extensions], [])
+        for dependency in dependencies:
+            handler_i = ExtHandlerInstance(dependency.handler, self.protocol)
+            timeout = 90 if dependency.timeout is None else dependency.timeout
+            timeout_delta = datetime.timedelta(seconds=(timeout * 60))
+            begin = datetime.datetime.utcnow()
+            for ext in dependency.exts:
+                status = None
+                while timeout_delta > (datetime.datetime.utcnow() - begin):
+                    status = handler_i.collect_ext_status(ext)
+                    if status is None:
+                        break
+                    all_success = status.status == "success"
+                    for substatus in status.substatusList:
+                        if substatus.status != "success":
+                            all_success = False
+                            break
+                    if all_success:
+                        break
+                    time.sleep(10)
```

Comment:
Should this be `time.sleep(timeout_delta)`? Maybe I am confused about `timeout` and `timeout_delta` - could the naming be clearer?

Reply:
`timeout` is an integer, `timeout_delta` is a timedelta... I've simplified so that `timeout` is an integer and then a timedelta.

10 seconds is some arbitrary number to avoid rereading the status blob too frequently. I am open to suggestions if you would prefer to make it more or less frequent (or to not sleep at all).